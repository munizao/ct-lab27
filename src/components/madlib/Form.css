.InputBox {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  margin: 30px 20vw 30px 20vw;
}

.ButtonBox {
  display: flex;
  justify-content: center;
  margin: 10px;
}

.Button {
  margin: 5px
}