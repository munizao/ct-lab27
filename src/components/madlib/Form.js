import React from 'react';
import PropTypes from 'prop-types';
import Input from './Input';
import styles from './Form.css';


const partsOfSpeech = ['adjective', 'noun', 'verb, past tense',
  'adverb', 'adjective', 'noun', 'noun',
  'adjective', 'verb', 'adverb', 'verb, past tense', 'adjective'];

const Form = ({ onSubmit, onChange, onClear, words }) => (
  <div>
    <div className={styles.InputBox}>
      {partsOfSpeech.map((pos, i) => <Input value={words[i]} key={i} placeholder={pos} onChange={onChange} index={i} />)}
    </div>
    <div className={styles.ButtonBox}>
      <button className={styles.Button} onClick={onSubmit}>Create Madlib</button>
      <button className={styles.Button} onClick={onClear}>Clear Form</button>
    </div>
  </div>
);

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onClear: PropTypes.func.isRequired,
  words: PropTypes.array.isRequired
};

export default Form;
