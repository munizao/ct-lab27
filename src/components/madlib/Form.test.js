import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form words={new Array(12).fill('')} onSubmit={() => {}} onChange={() => {}} onClear={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
