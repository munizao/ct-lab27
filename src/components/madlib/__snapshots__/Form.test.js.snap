// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <Input
      index={0}
      key="0"
      onChange={[Function]}
      placeholder="adjective"
      value=""
    />
    <Input
      index={1}
      key="1"
      onChange={[Function]}
      placeholder="noun"
      value=""
    />
    <Input
      index={2}
      key="2"
      onChange={[Function]}
      placeholder="verb, past tense"
      value=""
    />
    <Input
      index={3}
      key="3"
      onChange={[Function]}
      placeholder="adverb"
      value=""
    />
    <Input
      index={4}
      key="4"
      onChange={[Function]}
      placeholder="adjective"
      value=""
    />
    <Input
      index={5}
      key="5"
      onChange={[Function]}
      placeholder="noun"
      value=""
    />
    <Input
      index={6}
      key="6"
      onChange={[Function]}
      placeholder="noun"
      value=""
    />
    <Input
      index={7}
      key="7"
      onChange={[Function]}
      placeholder="adjective"
      value=""
    />
    <Input
      index={8}
      key="8"
      onChange={[Function]}
      placeholder="verb"
      value=""
    />
    <Input
      index={9}
      key="9"
      onChange={[Function]}
      placeholder="adverb"
      value=""
    />
    <Input
      index={10}
      key="10"
      onChange={[Function]}
      placeholder="verb, past tense"
      value=""
    />
    <Input
      index={11}
      key="11"
      onChange={[Function]}
      placeholder="adjective"
      value=""
    />
    <button>
      Create Madlib
    </button>
  </form>
  <button
    onClick={[Function]}
  >
    Clear Form
  </button>
</div>
`;
